### PR TITLE
DOC add sections to overview in scipy.stats

### PR DESCRIPTION
**Pull request by chrisb83**
The overview of the module `scipy.stats` is hard to read in my view: https://docs.scipy.org/doc/scipy/reference/stats.html. 

I tried to group the functions to give a better overview. This is an initial proposal, probably there are a lot of other ways to structure it, feedback welcome.

### CODE REVIEW

File: scipy/stats/__init__.py [hunk 1]
```diff
@@ -160,11 +160,12 @@
    skellam           -- Skellam
    zipf              -- Zipf
 
-Statistical functions
-=====================
+In the sequel, an overview of statistical functions is given.
+Several of these functions have a similar version in
+scipy.stats.mstats which work for masked arrays.
```
Comment (rgommers) [inline scipy/stats/__init__.py]: single backticks around `scipy.stats.mstats` should turn that into a link

File: scipy/stats/__init__.py [hunk 2]
```diff
@@ -160,11 +160,12 @@
    skellam           -- Skellam
    zipf              -- Zipf
 
-Statistical functions
-=====================
+In the sequel, an overview of statistical functions is given.
```
Comment (rgommers) [inline scipy/stats/__init__.py]: "In the sequel" sounds a bit odd to me, maybe just "Below"?

**Review by rgommers — CHANGES_REQUESTED** [review]
Good idea. Just a few minor comments

**Comment by chrisb83** [comment]
@rgommers done, i addressed your comments.

**Review by rgommers — APPROVED** [review]

**Comment by rgommers** [comment]
Merged, thanks @chrisb83

**Pull request merged**